crest/Assets/Crest/Crest/Scripts/Helpers/UnderwaterEffect.cs

```diff
@@ -158,6 +160,32 @@ private void LateUpdate()
             }
         }
 
+        bool ShowEffect()
+        {
+            if (_turnOffOutsideWaterBodies && OceanRenderer.Instance.WaterBodies.Count > 0)
+            {
+                var inOne = false;
+                float x = transform.position.x, z = transform.position.z;
+                foreach (var body in OceanRenderer.Instance.WaterBodies)
```

Comment:
Wouldn't having a function like "IsInsideWaterBody" be useful on the ocean renderer instance? As this pattern appears to crop-up a couple of times.

Reply:
yeah it could be, ill have a look

Reply:
i think i feel inclined to leave as is as the code is simple enough, and its executed ~100 times each frame so i'm not sure incurring the overhead of a function call is worth it. let me know if that sounds ok.

Reply:
Sure thing. Although it's worth pointing out that the C# compiler can inline functions if they are suitable for it and you can tell it to do so:

https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.methodimploptions?view=netcore-3.1

I believe this works with IL2CPP as Unity use it in a lot of their internal code.

Reply:
Thanks ill be lazy and go with whats there then, and keep this for future record. i think we'll need to review the cpu perf at some point which might be a good time to try this out and verify it doesnt hurt perf. (i could see this going to burst jobs as well)